code/socialDistribution/views.py

```diff
@@ -409,21 +418,29 @@ def editPost(request, id):
                 post.unlisted = form.cleaned_data.get('unlisted')
                 post.content_media = content_media
 
-                categories = form.cleaned_data.get('categories').split()
-                previousCategories = Category.objects.filter(post=post)
-                previousCategoriesNames = [
-                    cat.category for cat in previousCategories]
-
-                # Create new categories
-                for category in categories:
-                    if category in previousCategoriesNames:
-                        previousCategoriesNames.remove(category)
-                    else:
-                        Category.objects.create(category=category, post=post)
-
-                # Remove old categories that were deleted
-                for category in previousCategoriesNames:
-                    Category.objects.get(category=category, post=post).delete()
+                categories = form.cleaned_data.get('categories')
+
+                if categories is not None:
+                    categories = categories.split()
+                    categories_to_remove = [ cat.category for cat in post.categories.all()]
+
+                    """
+                    This implementation makes category names case-insensitive.
+                    This makes handling Category objects cleaner, albeit slightly more
+                    involved.
+                    """
+                    for category in categories:
+                        category_obj, created = Category.objects.get_or_create(
+                            category__iexact=category,
+                            defaults={'category': category}
+                        )
+                        post.categories.add(category_obj)
+                        
+                        while category_obj.category in categories_to_remove:
+                            categories_to_remove.remove(category_obj.category)     # don't remove this category
+
+                    for category in categories_to_remove:
+                        post.categories.get(category=category).delete()
```

Comment:
Does this delete the category or just remove it from the categories of this specific post?

Reply:
You're right. I should has used remove instead delete
I'll make the changes
Thank you!🙏